### DISCUSSION TITLE
Skip Lifecycle hook when publishing my manifest

### DISCUSSION THREAD

**Discussion by normj** [microsoft/aspire#2054] [Q&A] [answered]
In my PR for `Aspire.Hosting.AWS` I have a lifecycle hook to provision AWS resources. If the AppHost is being executed with `--publisher manifest` I don't want to run the lifecycle hook. Seems like I shouldn't be making any mutating changes when generating the manifest.

The problem I'm having is in the my AddXResource extension methods where I know to add the lifecycle I just have access to the `IDistributedApplicationBuilder` and that doesn't have any context of the publisher. From inside the lifecycle hook the only thing I have found is I could inject the `DistributedApplicationOptions` and then search the `Args` property for the command line arguments to tell if this is publisher manifest mode but that seems hacky. Is there a better way to not include the lifecycle hook when publishing a manifest that I'm not seeing?

**Comment by davidfowl** [comment, marked as answer]
https://github.com/dotnet/aspire/blob/a671e48abaa43eb75f9b05de5b586d2efe21f9e6/src/Aspire.Hosting.Azure.Provisioning/Provisioners/AzureProvisioner.cs#L41-L45

Less lacky but hacky still. Will be resolved by https://github.com/dotnet/aspire/issues/1823

**Reply by normj** [reply]
Thanks I'll add that to the PR.
